spec/emitter_spec.rb

```diff
@@ -729,9 +728,45 @@
         "thread=c at=start\n",
         "line=3\n",
         "thread=c at=finish elapsed=0.0000\n"
-      )
+      ) # the line "thread=e line=6" is not here b/c that was in a `#silence` block
 
       expect(other_lines).to contain_exactly("thread=d line=5\n")
     end
+
+    describe "#silence!" do
+      after do |example|
+        emitter.unsilence!
+      end
+
+      it "only affects the current thread" do
+        emitter.silence!
+
+        1_000.times do
+          emitter.log(thread: :main, line: 1)
+        end
+
+        thread_b = Thread.new {
+          1_000.times do
+            emitter.log(thread: :b, line: 2)
+          end
+        }
+
+        thread_c = Thread.new {
+          1_000.times do
+            emitter.log(thread: :c, line: 3)
+          end
```

Comment:
Is there a particular reason to have 2 extra threads?
Does the output changes if you remove thread_c?

Reply:
Good question. The extra threads are a [ham-fisted](https://www.merriam-webster.com/dictionary/ham-fisted) attempt to make sure we have enough happening that the thread scheduler will interleave some of the work. And then in a thread-unsafe setup (as we used to have) we'd see things break. Which is hard to prove since now the code is (we're pretty sure) thread safe. If you look at the other tread-related specs in the file you'll see they do something similar. 🤷 

I am 💯 game to clean this up and make it easier to understand. Thoughts?

Reply:
🤔 
Maybe moving the code to different methods would make it easier to understand, something like this:

```
    describe "#silence!" do
      it "only affects the current thread" do
        threads = do_worlk_on_unsilenced_thread

        threads << do_work_on_silenced_thread

        wait_for_threads(threads)

        verify_only_unsilenced_thread_emits_logs
      end
    end
```
Arggg testing multithread stuff is hard